File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals=sorted(intervals,key=lambda x:x[0])
        curr=intervals[0]
        res=[]
        for nextInterval in intervals:
            if nextInterval[0]<=curr[1]:
                if nextInterval[1]>curr[1]:
                    curr[1]=nextInterval[1]
            else:
                res.append(curr)
                curr=nextInterval
        res.append(curr)
        return res
            
                

File: 56-merge-intervals/56-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,(i1,i2)->Integer.compare(i1[0],i2[0]));
        List<int[]>res=new ArrayList<>();
        int curr[]=intervals[0];
        for(int[] nextinterval:intervals)
        {
            if(curr[1]>=nextinterval[0])
            {
                if(curr[1]<=nextinterval[1])
                {
                    curr[1]=nextinterval[1];
                }
            }
            else
            {
                res.add(curr);
                curr=nextinterval;
            }
        }
        res.add(curr);
        return res.toArray(new int[res.size()][]);
    }
}